testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -4028,6 +4028,9 @@ private void testDataMapping(DataMappingTestSetup dataMappingTestSetup)
         assertQuery("SELECT row_id FROM " + tableName + " WHERE value IS NULL OR value > " + sampleValueLiteral, "VALUES 'null value', 'high value'");
         assertQuery("SELECT row_id FROM " + tableName + " WHERE value IS NULL OR value <= " + highValueLiteral, "VALUES 'null value', 'sample value', 'high value'");
 
+        // complex condition, one that cannot be represented with a TupleDomain
+        assertQuery("SELECT row_id FROM " + tableName + " WHERE value = " + sampleValueLiteral + " OR another_column = " + sampleValueLiteral, "VALUES 'sample value'");
```

Comment:
Do we want to test whether we convert varchar containing backslash correctly to the remote database? Or do we want to test whether engine handles it correctly?

Appears the latter from the test (no pushdown so connector not involved).

Reply:
Got it. So we want to verify whether pushdown of values which contain special/escape characters work correctly - regardless of tuple domain based or ConnectorExpression pushdown.

This is useful context to add in commit msg.

Reply:
This was supposed to be convey via the code comment: "complex condition, one that cannot be represented with a TupleDomain"

please help me improve it

Reply:
added